src/species/corn.ts

```diff
@@ -2,6 +2,58 @@ import { Agent, BasicPlant, Species, Trait } from '../populations';
 
 const maturity = 300;
 
+const lifestage = {
+  seed: 0,
+  shoot: 1,
+  sprout: 2,
+  young: 3,
+  fullygrown: 4,
+  mature: 5,
+  flowering: 6,
+  harvested: 7
+};
+const lifestageThresholds = {
+  seed: 10,
+  shoot: maturity * 0.2,
+  sprout: maturity * 0.4,
+  young: maturity * 0.6,
+  fullygrown: maturity * 0.8,
+  mature: maturity,
+  flowering: maturity * 1.2,
+  harvested: maturity * 2,
+};
+
+const getLifestage = (agent: Agent): number => {
+  const age = agent.get('age');
+  let stage = lifestage.seed;
+
+  if (age < lifestageThresholds.shoot) {
+    stage = lifestage.seed;
+  }
+  if (age >= lifestageThresholds.shoot && age < lifestageThresholds.sprout) {
```

Comment:
A minor point which I forgot to mention on the worms PR is that I think the >= test is redundant here, so you could just have a sequence of < tests, which would simplify the code a bit.